Source/WebCore/page/Frame.cpp

```diff
@@ -257,4 +258,27 @@ void Frame::setOwnerElement(HTMLFrameOwnerElement* element)
     }
 }
 
+void Frame::setOwnerPermissionsPolicy(OwnerPermissionsPolicyData&& ownerPermissionsPolicy)
+{
+    // If the frame is in the same process as parent, it should compute policy from owner element directly.
+    if (dynamicDowncast<RemoteFrame>(tree().parent()))
+        m_ownerPermisssionsPolicy = WTFMove(ownerPermissionsPolicy);
+}
+
+std::optional<OwnerPermissionsPolicyData> Frame::ownerPermissionsPolicy()
+{
+    if (m_ownerPermisssionsPolicy)
+        return m_ownerPermisssionsPolicy;
+
+    RefPtr owner = ownerElement();
+    if (!owner)
+        return std::nullopt;
+
+    auto documentOrigin = WebCore::SecurityOriginData { owner->document().securityOrigin().data() };
```

Comment:
```suggestion
    auto documentOrigin = owner->document().securityOrigin().data();
```
No need for WebCore:: also